packages/webchannel-wrapper/gulpfile.js

```diff
@@ -0,0 +1,188 @@
+  'goog.events.CAPTURE_SIMULATION_MODE=0',
+  // Disable IE-Specific ActiveX fallback for XHRs (saves 524 bytes).
+  'goog.net.XmlHttpDefines.ASSUME_NATIVE_XHR=true'
+];
+
+/**
+ * Generates a closure compiler build of webchannel-wrapper.
+ * @param {string} filename name of the generated file
+ * @param {string} prefix prefix to the compiled code
+ * @param {string} suffix suffix to the compiled code
+ */
+function createBuildTask(filename, prefix, suffix) {
+  return function closureBuild() {
+    return gulp
+      .src(
+        [
+          `${closureLibRoot}/closure/goog/**/*.js`,
+          `${closureLibRoot}/third_party/closure/goog/**/*.js`,
+          'src/**/*.js'
+        ],
+        { base: '.' }
+      )
+      .pipe(sourcemaps.init())
+      .pipe(
+        closureCompiler({
+          js_output_file: filename,
+          output_wrapper: `${prefix}%output%${suffix}`,
+          entry_point: 'firebase.webchannel.wrapper',
+          compilation_level: OPTIMIZATION_LEVEL,
+          externs: [
+            resolve(__dirname, './externs/overrides.js'),
+            resolve(__dirname, './externs/module.js')
+          ],
+          language_out: 'ECMASCRIPT_2017',
+          dependency_mode: 'PRUNE',
+          define: closureDefines
+        })
+      )
+      .pipe(sourcemaps.write('.'))
+      .pipe(gulp.dest('dist'));
+  };
+}
+
+function createRollupTask({
+  inputPath,
+  outputExtension,
+  compileToES5 = false,
+  format = 'es'
+}) {
+  return async function rollupBuild() {
+    const plugins = [rollupSourcemaps(), commonjs()];
+    if (compileToES5) {
+      plugins.push(
+        typescriptPlugin({
+          typescript,
+          // Uncomment for development only. Prevents caching between builds.
+          // clean: true,
+          compilerOptions: { allowJs: true },
+          include: ['dist/**/*.js']
+        })
+      );
+    }
+    // Typescript step unminifies minified Closure output.
+    plugins.push(terser());
+    const inputOptions = {
+      input: inputPath,
+      plugins,
+      external: ['@firebase/app']
+    };
+
+    const outputOptions = {
+      file: `dist/index${outputExtension ? '.' : ''}${outputExtension}.js`,
+      format,
+      sourcemap: true,
+      exports: 'named'
+    };
+
+    const bundle = await rollup.rollup(inputOptions);
+    return bundle.write(outputOptions);
+  };
+}
+
+async function deleteIntermediateFiles() {
+  await del('dist/temp');
+}
+
+// Closure-generated ES2017 intermediate file (CJS format)
+const intermediateCjsFile = 'temp/cjs.js';
+const intermediateCjsPath = resolve(__dirname, 'dist/', intermediateCjsFile);
+const cjsBuild = createBuildTask(
+  intermediateCjsFile,
+  CJS_WRAPPER_PREFIX,
+  CJS_WRAPPER_SUFFIX
+);
+
+// Closure-generated ES2017 intermediate file (no wrapper text)
+const intermediateEsmFile = 'temp/esm.js';
+const intermediateEsmPath = resolve(__dirname, 'dist/', intermediateEsmFile);
+const esmBuild = createBuildTask(intermediateEsmFile, '', '');
```

Comment:
Does sourcemap work correctly?
I think you need to add a suffix in order to pipe the sourcemap generated by closure into rollup. `//# sourceMappingURL=esm.js.map',`. You can take a look at how it is done currently.

Reply:
`gulp-sourcemaps` seems to add this line as part of its process.  If I add it to the suffix I see it twice in the closure output file.

Reply:
SG